### PR TITLE
feat(network): implement blocklist behaviour

### PR DESCRIPTION
**Pull request by RolandSherwin**
- uses `libp2p::allow_block_list` Behavior to deny connection to a bad node.

### CODE REVIEW

File: sn_networking/src/driver.rs [hunk 1]
```diff
@@ -184,20 +184,24 @@ pub enum VerificationKind {
...
-    pub(super) request_response: request_response::cbor::Behaviour<Request, Response>,
-    pub(super) kademlia: kad::Behaviour<UnifiedRecordStore>,
+    pub(super) blocklist:
+        libp2p::allow_block_list::Behaviour<libp2p::allow_block_list::BlockedPeers>,
+    pub(super) identify: libp2p::identify::Behaviour,
     #[cfg(feature = "local-discovery")]
     pub(super) mdns: mdns::tokio::Behaviour,
-    pub(super) identify: libp2p::identify::Behaviour,
-    pub(super) dcutr: libp2p::dcutr::Behaviour,
+    #[cfg(feature = "upnp")]
```
Comment (joshuef) [inline sn_networking/src/driver.rs:198]: why upnp only?

Comment (RolandSherwin) [reply sn_networking/src/driver.rs:198]: I just reordered the behaviours here, there is no actual change.

File: sn_networking/src/event/swarm.rs [hunk 2]
```diff
@@ -137,7 +137,8 @@ impl SwarmDriver {
                                 our_protocol: IDENTIFY_PROTOCOL_STR.to_string(),
                                 their_protocol: info.protocol_version,
                             });
-                            self.remove_outdated_connections(Some(peer_id));
+                            // Block the peer from any further communication.
+                            self.swarm.behaviour_mut().blocklist.block_peer(peer_id);
```
Comment (maqi) [inline sn_networking/src/event/swarm.rs:141]: so the already established connection will be disconnected automatically, once the peer is in the blocklist?
i.e. no more explicit `remove_connection` call to that peer is needed ?

File: sn_networking/src/driver.rs (hunk 1 above)
Comment (maqi) [reply sn_networking/src/driver.rs:198]: yeah, that toggle is only for upnp, and just being moving place for a better ordering

File: sn_networking/src/event/swarm.rs (hunk 2 above)
Comment (RolandSherwin) [reply sn_networking/src/event/swarm.rs:141]: That is right! It internally calls `ToSwarm::CloseConnection`

**Review by maqi — APPROVED** [review]

**Pull request merged**